tensorflow/c/tf_tensor.h

```diff
@@ -45,6 +45,24 @@ limitations under the License.
 extern "C" {
 #endif
 
+// Macro used to calculate struct size for maintaining ABI stability across 
+// different struct implementations. 
+#ifndef TF_OFFSET_OF_END
+#define TF_OFFSET_OF_END(TYPE, MEMBER) (offsetof(TYPE, MEMBER) + \
+    sizeof(((TYPE *)0)->MEMBER))
+#endif // TF_OFFSET_OF_END
+
+// Allocator Attributes used for tensor allocation. 
+typedef struct TF_AllocatorAttributes { 
+  size_t struct_size; 
+  // Set boolean to 1 for CPU allocation, else 0. 
+  unsigned char on_host; 
+} TF_AllocatorAttributes; 
```

Comment:
Also, would it make sense to have a separate header for this type?

Reply:
We can make it a separate header. @alextp, Alex, as API owner, do you think it is better to have a different header for TF_AllocatorAttributes?

Reply:
I don't think we need a new header now.